lib/utils.js

```diff
@@ -26,7 +26,8 @@ function isArray(val) {
  * @returns {boolean} True if value is a Buffer, otherwise false
  */
 function isBuffer(val) {
-  return ![undefined, null].includes(val) && val.constructor === Buffer;
+  if (val === null || val === void 0 || val.constructor === null || val.constructor === void 0 )  return false;
```

Comment:
what's the difference between using `void 0` and undefined?

Reply:
`undefined` is not a reserved word in javascript, so it can be rewritten. `void 0` is stricter.

Reply:
we rely on `type of x === 'undefined'` throughout the file, might be better to be consistent, also are there cases where the constructor can be null? could it be sufficient to just check `!isUndefined(val.constructor)`, also we can just use our own helper? `isUndefined`

Reply:
I agree to use `isUndefined`, and I suggest to replace all the other `undefined` judgement with our own helper function `isUndefined`, if you agree, then I will submit a PR.

if `constructor` has been overwriten with `null` (extreme paticular case), then `typeof val.constructor.isBuffer === 'function'` will report an error.